barman/output.py

```diff
@@ -1503,117 +1582,157 @@ def result_list_backup(self, backup_info, backup_size, wal_size, retention_statu
     def result_show_backup(self, backup_ext_info):
         """
         Output all available information about a backup in show-backup command
+        in json format.
 
-        The argument has to be the result
-        of a Server.get_backup_ext_info() call
+        The argument has to be the result of a Server.get_backup_ext_info() call
 
         :param dict backup_ext_info: a dictionary containing
             the info to display
         """
         data = dict(backup_ext_info)
+
+        # Json top layer info
         server_name = data["server_name"]
 
+        # Json second layer info
         output = self.json_output[server_name] = dict(
             backup_id=data["backup_id"], status=data["status"]
         )
-
-        if "backup_name" in data and data["backup_name"] is not None:
+        # Json second layer info
+        # Server information
+        output["server_information"] = dict(
+            data_checksums=data["data_checksums"],
+            summarize_wal=(
+                data["summarize_wal"] if data["summarize_wal"] else "not supported"
+            ),
+        )
+        if "backup_name" in data and data["backup_name"]:
             output.update({"backup_name": data["backup_name"]})
 
+        if "systemid" in data and data["systemid"]:
+            output.update({"system_id": data["systemid"]})
+
         if data["status"] in BackupInfo.STATUS_COPY_DONE:
+            # Json second layer info
+            # General information
             output.update(
                 dict(
                     postgresql_version=data["version"],
                     pgdata_directory=data["pgdata"],
-                    data_checksums=data["data_checksums"],
+                    cluster_size=data["cluster_size"],
                     tablespaces=[],
                 )
             )
-            if "snapshots_info" in data and data["snapshots_info"]:
-                output["snapshots_info"] = data["snapshots_info"]
-            if data["tablespaces"]:
-                for item in data["tablespaces"]:
-                    output["tablespaces"].append(
-                        dict(name=item.name, location=item.location, oid=item.oid)
-                    )
-
+            # Json second layer info
+            # Base Backup information
             output["base_backup_information"] = dict(
-                disk_usage=pretty_size(data["size"]),
```

Comment:
If you think about it this would be a "breaking change".
In theory, per semantic versioning the next release should be a major instead of a minor, if this gets in.
Thoughts @martinmarques ?